app/MwApi/MwApiExtras.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace App\MwApi;
+
+use Mediawiki\Api\SimpleRequest;
+
+/**
+ * Extra functions that addwiki/mediawiki-api does not provide itself
+ */
+class MwApiExtras
+{
+    public static function canEmail($wiki, $username)
+    {
+        $api = MwApiGetter::getApiForWiki($wiki);
+        $response = $api->getRequest(new SimpleRequest(
+            'query',
+            [
+                'list' => 'users',
+                'ususers' => $username,
+                'usprop' => 'emailable',
+            ]
+        ));
+
+        return !empty($response['query']['users']) && isset($response['query']['users'][0]['emailable']);
+    }
+
+    public static function sendEmail($wiki, $username, $title, $content)
+    {
+        $api = MwApiGetter::getApiForWiki($wiki);
+
+        $response = $api->postRequest(new SimpleRequest('emailuser', [
+            'token' => $api->getToken(),
+            'target' => $username,
+            'subject' => $title,
+            'text' => $content,
+        ]));
+
+        return $response['emailuser']['result'] === 'Success';
+    }
+
+    public static function getBlockInfo($wiki, $username)
+    {
+        $key = filter_var($username, FILTER_VALIDATE_IP) === false ? 'bkusers' : 'bkip';
```

Comment:
While it's not widely advertised, autoblocks can be appealed, hense the bkids presence in wikiactions.py - so we need to look at adding this